packages/psss/src/api/queries/useUpcomingReport.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { getCurrentPeriod } from '@tupaia/utils';
+import { useQuery } from 'react-query';
+import { subtractWeeksFromPeriod, getDaysTillDueDay } from '../../utils';
+import { get } from '../api';
+import { REPORT_STATUSES } from '../../constants';
+
+export const useUpcomingReport = countryCode => {
+  const currentPeriod = getCurrentPeriod('WEEK');
+  const lastPeriod = subtractWeeksFromPeriod(currentPeriod, 1);
+  const endpoint = `confirmedWeeklyReport/${countryCode.toUpperCase()}`;
+  const query = useQuery(
+    [endpoint, lastPeriod],
+    () => get(endpoint, { params: { startWeek: lastPeriod, endWeek: lastPeriod } }),
+    { staleTime: 60 * 1000 },
+  );
+
+  const data = query?.data?.data?.results ?? [];
+
+  const { isLoading } = query;
+
+  if (isLoading) {
+    return query;
+  }
+
+  const isConfirmed = data.length > 0;
+
+  const days = getDaysTillDueDay();
+
+  if (isConfirmed) {
+    const totalDays = days + 7;
+
+    return {
+      ...query,
+      data,
+      reportStatus: REPORT_STATUSES.UPCOMING,
+      period: currentPeriod,
+      days: totalDays,
+    };
+  }
+
+  const status = days > 0 ? REPORT_STATUSES.UPCOMING : REPORT_STATUSES.OVERDUE;
+
+  return {
+    ...query,
+    data,
+    reportStatus: status,
+    period: lastPeriod,
+    days,
```

Comment:
Perhaps using `daysTillDueDay` instead of `days` would be even clearer (while the latter is pretty clear here, I was a bit confused when confirming its usage in `UpcomingReportCard`